plugins/woocommerce-admin/client/products/add-product-page.tsx

```diff
@@ -3,23 +3,13 @@
  */
 import { recordEvent } from '@woocommerce/tracks';
 import { useEffect } from '@wordpress/element';
-import { Form } from '@woocommerce/components';
-import { Product } from '@woocommerce/data';
 
 /**
  * Internal dependencies
  */
-import { ProductFormHeader } from './layout/product-form-header';
-import { ProductFormLayout } from './layout/product-form-layout';
-import { ProductDetailsSection } from './sections/product-details-section';
-import { ProductInventorySection } from './sections/product-inventory-section';
-import { PricingSection } from './sections/pricing-section';
-import { ProductShippingSection } from './sections/product-shipping-section';
-import { ImagesSection } from './sections/images-section';
+
 import './product-page.scss';
-import { validate } from './product-validation';
-import { AttributesSection } from './sections/attributes-section';
-import { ProductFormFooter } from './layout/product-form-footer';
+import { ProductForm } from './product-form';
 
 const AddProductPage: React.FC = () => {
```

Comment:
Could we merge `AddProductPage` and `EditProductPage` into a single component `ProductPage`?  I think we could use `const { productId } = useParams();` to distinguish whether or not we're creating a new product or editing one.

Reply:
@nathanss attempted this but we ran into 2 issues:

1. `getPermalinkParts` did not seem to get the parts correctly when switching between pages without a full refresh.  Possibly related to encoding of slashes in the URL but I'm not entirely sure.  cc @louwie17 if you have any context around this
2. React router did not re-render properly when switching from a created product to a new product form.  This may be due to them sharing the same component and caching happening on the routing level.

Since the remainder of the logic has been merged in this PR, I don't think we should let this block us from merging this one.